TMS.Net07.Homework4.2FibonachiFactorial/TMS.Net07.Homework4.2FibonachiFactorial/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TMS.Net07.Homework4._2FibonachiFactorial
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string oper;
+            Console.WriteLine("Выбор операции : n!,  n!2,  f(n), ");
+            oper = Console.ReadLine();
+            Console.WriteLine("Введите значение : ");
+
+            switch (oper)
+            {
+                case "n!":
+                    int value = Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Факториал числа {0} равен {1}", value, Factorial(value));
+                    break;
+                case "n!2":
+                    int num = Convert.ToInt32(Console.ReadLine());
+                    Factorial2(num);
+                    break;
+                case "f(n)":
+                    int fibo = Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("{0}", Fibonachi(fibo));
+                    break;
+                default:
+                    Console.WriteLine("Error");
+                    break;
+            }
+            Console.ReadKey();
+        }
+        //рекурсивный рассчет факториала числа
+        static int Factorial(int value)
+        {
+            if (value == 0)
```

Comment:
`-1` is also a number 😉
We risque to get into an infinite loop, because the `Factorial` method stops recursion only when number becomes `0`, but in the case of initial `-1` it'll never happen